EIPS/eip-7791.md

```diff
@@ -36,18 +36,24 @@ This helps align the incentives of smart contract authors, validators, and the b
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
+A new quantity is added to the calling context, the `gas2eth` limit. It is analogous to the gas counter, but is transaction-global and is maintained separately from the gas limit. At the beginning of the transaction, it is initialized to the same value as the tx gas limit.
+
 A new opcode is introduced, `GAS2ETH` (`0xFC`), which:
 
 - Pops two values from the stack: `addr` then `gas_amount`. If there are fewer than two values on the stack, the calling context should fail with stack underflow.
-- Deducts `gas_amount` from the current calling context.
+- Deducts `gas_amount` from the current calling context's `gas2eth` limit.
 - Computes `wei_val` by multiplying `gas_amount` by the current transaction context's `gas_price` ([EELS][gasprice]).
 - Endows the address `addr` with `wei_val` wei.
-- If the gas cost of this opcode + `gas_amount` is greater than the available gas in the current calling context, the calling context should fail with "out-of-gas" and any state changes reverted.
+- If `gas_amount` is greater than the remaining `gas2eth` limit, the calling context should fail with "out-of-gas" and any state changes reverted.
 - Pushes `wei_val` onto the stack.
 
+On reverts (e.g., through the `REVERT` opcode or invalid opcodes), any gas paid out via `GAS2ETH` should be returned to the gas2eth budget.
```

Comment:
we have _multiple_ invalid opcodes now? Also, what about exception halts?

Reply:
i'll change to "or exceptional halts" to be more general though